pkg/chart/v2/util/coalesce.go

```diff
@@ -301,7 +301,7 @@ func coalesceTablesFullKey(printf printFn, dst, src map[string]interface{}, pref
 			} else {
 				printf("warning: cannot overwrite table with non table for %s (%v)", fullkey, val)
 			}
-		} else if istable(dv) && val != nil {
+		} else if istable(dv) && val != nil && isNonEmptyString(val) && isNonEmptyTable(dv) {
 			printf("warning: destination for %s is a table. Ignoring non-table value (%v)", fullkey, val)
```

Comment:
These prints should be slog calls.

Should we have a warning more specific to the condition?

Reply:
> Should we have a warning more specific to the condition?

I think you're asking if the additional conditionals means we should change the text of the warning. I would say no. I think the wording of the warning matches the conditionals now, and actually did not match the conditional  prior to this PR, since the warning would be printed unnecessarily.

If we're adjusting the wording though, I'd suggest removing the use of the word "table" as helm end-users don't know what a table is.

```suggestion
			printf("warning: overridden setting %s is a table. Ignoring non-matching data type (%v)", fullkey, val)
```

Reply:
> These prints should be slog calls.

I agree. It's a little strange how log.Printf is passed into this function. I'm also finding that there is now a `internal/chart/v3/util/coalesce.go` AND a `pkg/chart/v2/util/coalesce.go` (wasn't here the last time I updated this PR). Perhaps changing the logging + conditional should only be done in the v3 version.